### ISSUE TITLE
【RFC】useNetwork

### ISSUE THREAD

**Issue by Straw94** [alibaba/hooks#512]
## API
```javascript
interface INetworkState {
  online?: boolean;
  since?: Date;
  downlink?: number;
  downlinkMax?: number;
  effectiveType?: string;
  rtt?: number;
  type?: string;
}

export function useNetwork(
  initialState: INetworkState | (() => INetworkState)
): INetworkState
```

## DEMO
```javascript
import { useNetwork } from 'ahooks';

const Demo = () => {
  const networkState = useNetwork();
  return (
    <pre>
      {JSON.stringify(networkState, null, 2)}
    </pre>
  );
};
```

**Comment by brickspert** [comment]
+1

**Comment by awmleer** [comment]
Implemented in pull request #569 .
